tfx/examples/imdb/imdb_utils_native_keras.py

```diff
@@ -0,0 +1,206 @@
+      dataset to combine in a single batch.
+
+  Returns:
+    A dataset that contains (features, indices) tuple where features is a
+      dictionary of Tensors, and indices is a single Tensor of label indices.
+  """
+  transformed_feature_spec = (
+      tf_transform_output.transformed_feature_spec().copy())
+
+  dataset = tf.data.experimental.make_batched_features_dataset(
+      file_pattern=file_pattern,
+      batch_size=batch_size,
+      features=transformed_feature_spec,
+      reader=_gzip_reader_fn,
+      label_key=_LABEL_KEY)
+
+  return dataset
+
+def _build_keras_model() -> keras.Model:
+  """Creates a DNN Keras model for classifying imdb data.
+
+  Returns:
+    A Keras Model.
+  """
+  # The model below is built with Functional API, please refer to
+  # https://www.tensorflow.org/guide/keras/overview for all API options.
+  model = keras.Sequential([
+      keras.layers.Embedding(
+          _MAX_FEATURES+1,
+          _HIDDEN_UNITS,
+          mask_zero=True),
+      keras.layers.Bidirectional(keras.layers.LSTM(
+          _HIDDEN_UNITS,
+          dropout=_DROPOUT_RATE,
+          recurrent_dropout=_DROPOUT_RATE)),
+      keras.layers.Dense(1, activation='sigmoid')
+  ])
+
+  model.compile(
+      loss='binary_crossentropy',
+      optimizer=keras.optimizers.Adam(_LEARNING_RATE),
+      metrics=['AUC', 'binary_accuracy'])
+
+  model.summary()
+  return model
+
+def _get_serve_tf_examples_fn(model, tf_transform_output):
+  """Returns a function that parses a serialized tf.Example."""
+  model.tft_layer = tf_transform_output.transform_features_layer()
+
+  @tf.function
+  def serve_tf_examples_fn(serialized_tf_examples):
+    """Returns the output to be used in the serving signature."""
+    feature_spec = tf_transform_output.raw_feature_spec()
+    feature_spec.pop(_LABEL_KEY)
+    parsed_features = tf.io.parse_example(serialized_tf_examples, feature_spec)
+
+    transformed_features = model.tft_layer(parsed_features)
+    # TODO(b/148082271): Remove this line once TFT 0.22 is used.
+    transformed_features.pop(_LABEL_KEY, None)
+
+    return model(transformed_features)
+
+  return serve_tf_examples_fn
+
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+  """Train the model based on given args.
+
+  Args:
+    fn_args: Holds args used to train the model as name/value pairs.
+  """
+  tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+
+  train_dataset = _input_fn(
+      fn_args.train_files,
+      tf_transform_output,
+      batch_size=_TRAIN_BATCH_SIZE)
+
+  eval_dataset = _input_fn(
+      fn_args.eval_files,
+      tf_transform_output,
+      batch_size=_EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model = _build_keras_model()
+
+  model.fit(
+      train_dataset,
+      epochs=_TRAIN_EPOCHS,
+      steps_per_epoch=fn_args.train_steps,
```

Comment:
Lets follow the following example of how to map train_steps into epochs and how to use steps_per_epoch:
https://github.com/tensorflow/tfx/blob/2c1d9561a79caeb47f9964c2822b0dd8d13d942d/tfx/examples/iris/iris_utils_native_keras.py#L239

Reply:
would this run into this error?
`WARNING:tensorflow:Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches (in this case, 200 batches). You may need to use the repeat() function when building your dataset.`

the kera.model.fit take epoch as a pass through the entire dataset.

Reply:
This is not an issue. As the dataset generated by csvExampleGen should have repeat=True